code/game/objects/items/devices/traitordevices.dm

```diff
@@ -168,3 +168,204 @@ effective or pretty fucking useless.
 		GLOB.active_jammers |= src
 	else
 		GLOB.active_jammers -= src
+
+/obj/item/teleporter
+	name = "Syndicate teleporter"
+	desc = "A strange syndicate version of a cult veil shifter. Warrenty voided if exposed to EMP."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "syndi-tele"
+	throwforce = 5
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 4
+	throw_range = 10
+	flags = CONDUCT
+	item_state = "electronic"
+	origin_tech = "magnets=3;combat=3;syndicate=3"
+	var/tp_range = 8
+	var/inner_tp_range = 3
+	var/charges = 4
+	var/max_charges = 4
+	var/saving_throw_distance = 3
+	var/flawless = FALSE
+
+/obj/item/teleporter/Initialize(mapload, ...)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/teleporter/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] has [charges] out of [max_charges] charges left.</span>"
+
+/obj/item/teleporter/attack_self(mob/user)
+	attempt_teleport(user, FALSE)
+
+/obj/item/teleporter/process()
+	if(prob(10) && charges != max_charges)
+		charges += 1
```

Comment:
```suggestion
	if(prob(10) && charges < max_charges)
		charges++
```
Styling + This prevents it being infinitely overcharged if it somehow goes above